src/eins-hwinfo.c

```diff
@@ -454,22 +364,107 @@ eins_hwinfo_get_cpu_info (void)
   return eins_hwinfo_parse_lscpu_json (json_data, (gssize) json_size);
 }
 
+GVariant *
+eins_hwinfo_get_computer_hwinfo (void)
+{
+  guint32 ramsize = eins_hwinfo_get_ram_size ();
+  DiskSpaceType diskspace = {};
+  GVariant *cpuinfo = eins_hwinfo_get_cpu_info();
```

Comment:
This function can fail and return `NULL`. Passing `NULL` to `g_variant_new()` below will crash. This function should either gracefully return `NULL` in that case, or gracefully pass a GVariant  of an empty `a(squ)` array to `g_variant_new()` below. I think passing a new empty array is better since then we can at least report the ram size & disk space.

Reply:
sure